x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -104,6 +110,40 @@ static DeprecationIssue checkUniqueRealmOrders(final Settings settings, final Pl
         );
     }
 
+    static DeprecationIssue checkImplicitlyDisabledNativeRealms(final Settings settings, final PluginsAndModules pluginsAndModules) {
+        final Map<RealmConfig.RealmIdentifier, Settings> realmSettings = RealmSettings.getRealmSettings(settings);
+        if (realmSettings.isEmpty()) {
+            return null;
+        }
+        // If all configured realms are disabled, this equals to no realm is configured. The implicitly behaviour in this case
+        // is to add file and native realms. So we are good here.
```

Comment:
Are we good?
What will this config do in 8.0? 
```
xpack.security.authc.realms:
  native.native_realm:
    enabled: false
  ldap.corp_ldap:
    enabled: false
```

In 7.x it will result in both a file and native realm enabled (I think)
I believe the proposal is that in 8.x it will result in just a file realm.

Should we issue a deprecation warning (because the behaviour will change) in this case too?  
(A separate PR is fine though)

Reply:
Good catch. Yes this is indeed the proposal in the other PR #69096. I updated this PR to issue warning for this scenario as well. I'd like to mention that this change also means that the cluster will not automatically fallback to file/native realms when:
* The License is downgraded and
* The file or native realm is explicitly disabled

I don't know whether users rely on this "feature". I personally feel it is not something we should cater for because implicitly auto-enabling realms on the fly could be a security issue and also explicit behaviours are what we are after with these changes. All in all, if a realm is explicity disabled, I prefer it remain disabled at all times.